### PR TITLE
NBL-4: Node.js thin client: affinity awareness feature

### PR DESCRIPTION
**Pull request by ragrus-nbl**
What was done:
- Implementation of the Affinity Awareness feature
- Tests for this feature
- A separate test configuration

TODO:
- Check that jsdoc generates with no issues
- Make an updated instruction for tests (?)

### CODE REVIEW

File: modules/platforms/nodejs/lib/internal/Router.js [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+const BinaryUtils = require('./BinaryUtils');
+const BinaryObject = require('../BinaryObject');
+const ObjectType = require('../ObjectType');
+const ArgumentChecker = require('./ArgumentChecker');
+const Logger = require('./Logger');
+
+// Number of tries to get cache partitions info
+const GET_CACHE_PARTITIONS_RETRIES = 3;
+// Delay between tries to get cache partitions info
+const GET_CACHE_PARTITIONS_DELAY = 100;
```
Comment (isapego) [inline modules/platforms/nodejs/lib/internal/Router.js]: Is it in millis?

Comment (ragrus-nbl) [reply modules/platforms/nodejs/lib/internal/Router.js]: Yes, it is. I will add this to the comment for this const.

File: modules/platforms/nodejs/lib/internal/BinaryCommunicator.js [hunk 2]
```diff
@@ -51,8 +51,8 @@ class BinaryCommunicator {
         }
     }
 
-    async send(opCode, payloadWriter, payloadReader = null) {
-        await this._socket.send(opCode, payloadWriter, payloadReader);
+    async send(opCode, payloadWriter, payloadReader = null, affinityHint = null) {
+        await this._socket.send(opCode, payloadWriter, payloadReader, affinityHint);
```
[the remark this answers is not in the document]
Comment (ragrus-nbl) [reply modules/platforms/nodejs/lib/internal/BinaryCommunicator.js]: Right! Will rename it.

**Comment by ragrus-nbl** [comment]
> Also, please, make sure you insert ticket name prior to PR name as follows
> `NBL-4: Node.js thin client: affinity awareness feature`.
> This is important for automation

Yes, sure

**Pull request closed without merging**